pvlib/irradiance.py

```diff
@@ -2885,3 +2921,43 @@ def dni(ghi, dhi, zenith, clearsky_dni=None, clearsky_tolerance=1.1,
             (zenith < zenith_threshold_for_zero_dni) &
             (dni > max_dni)] = max_dni
     return dni
+
+
+def adjust_direct_for_horizon(poa_direct, horizon_profile,
+                              solar_azimuth, solar_zenith):
+    '''
+    Adjusts modeled DNI to account for the presence of a horizon. At
+    each time step check to see if the sun is behind the horizon. If so,
+    set the DNI to zero.
+
+    Parameters
+    ----------
+    poa_direct : numeric
+        The modeled direct normal irradiance in the plane of array.
+    solar_zenith : numeric
+        Solar zenith angle.
+    solar_azimuth : numeric
+        Solar azimuth angle.
+    horizon_profile : list
+        A list of (azimuth, dip_angle) tuples that defines the horizon
+        profile given. Every azimuth from -180 to 180, in 1 degree
+        increments, must be in this list.
+
```

Comment:
If you require 361 values, you do not need the tuples; just a vector of dip angles would do.

Reply:
ran with this